File: app/src/main/java/com/estrrado/tradopay/ui/modules/home/adapter/HomeDetailedrRoundItemAdapter.java
package com.estrrado.tradopay.ui.modules.home.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.estrrado.tradopay.ui.modules.OnItemSelectedlistner;
import com.example.tradopay.R;

public class HomeDetailedrRoundItemAdapter extends RecyclerView.Adapter<HomeDetailedrRoundItemAdapter.HomeDetailedrRoundItemViewHolder> {

    OnItemSelectedlistner onItemSelectedlistner;
    String[] arr;int[] imgs;
    public HomeDetailedrRoundItemAdapter(OnItemSelectedlistner onItemSelectedlistner, String[] headings_detail, int[] images_detail) {
        this.onItemSelectedlistner=onItemSelectedlistner;
        this.arr=headings_detail;
        this.imgs=images_detail;
    }

    @NonNull
    @Override
    public HomeDetailedrRoundItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new HomeDetailedrRoundItemViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_home_detail_item_round,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull HomeDetailedrRoundItemViewHolder holder, int position) {
        holder.img.setImageResource(imgs[position]);
        holder.txt.setText(arr[position]);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onItemSelectedlistner.onItemClick(null,position);
            }
        });

    }

    @Override
    public int getItemCount() {
        return arr.length;
    }

    public class HomeDetailedrRoundItemViewHolder extends RecyclerView.ViewHolder
    {
        TextView txt;
        ImageView img;
        public HomeDetailedrRoundItemViewHolder(@NonNull View itemView) {
            super(itemView);
            txt=itemView.findViewById(R.id.txtItemName);
            img=itemView.findViewById(R.id.imgDetailRound);
        }
    }
}

File: app/src/main/java/com/estrrado/tradopay/adapter/HomeDetailedrRoundItemAdapter.kt
package com.estrrado.tradopay.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.estrrado.tradopay.adapter.HomeDetailedrRoundItemAdapter.HomeDetailedrRoundItemViewHolder
import com.estrrado.tradopay.listener.OnItemSelectedlistner
import com.example.tradopay.R

class HomeDetailedrRoundItemAdapter(var onItemSelectedlistner: OnItemSelectedlistner, var arr: Array<String>, var imgs: IntArray) : RecyclerView.Adapter<HomeDetailedrRoundItemViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HomeDetailedrRoundItemViewHolder {
        return HomeDetailedrRoundItemViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.layout_home_detail_item_round, parent, false))
    }

    override fun onBindViewHolder(holder: HomeDetailedrRoundItemViewHolder, position: Int) {
        holder.img.setImageResource(imgs[position])
        holder.txt.text = arr[position]
        holder.itemView.setOnClickListener { onItemSelectedlistner.onItemClick(null, position) }
    }

    override fun getItemCount(): Int {
        return arr.size
    }

    inner class HomeDetailedrRoundItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var txt: TextView
        var img: ImageView

        init {
            txt = itemView.findViewById(R.id.txtItemName)
            img = itemView.findViewById(R.id.imgDetailRound)
        }
    }
}